tez-api/src/main/java/org/apache/tez/dag/api/TezConfiguration.java

```diff
@@ -396,8 +396,18 @@ public TezConfiguration(boolean loadDefaults) {
   @ConfigurationProperty
   public static final String TEZ_AM_LAUNCH_CLUSTER_DEFAULT_CMD_OPTS =
       TEZ_AM_PREFIX + "launch.cluster-default.cmd-opts";
-  public static final String TEZ_AM_LAUNCH_CLUSTER_DEFAULT_CMD_OPTS_DEFAULT =
-      "-server -Djava.net.preferIPv4Stack=true -Dhadoop.metrics.log.level=WARN";
+  public static final String TEZ_AM_LAUNCH_CLUSTER_DEFAULT_CMD_OPTS_DEFAULT;
+
+  static {
+    if (TezCommonUtils.getJavaVersion() >= 9) {
+      TEZ_AM_LAUNCH_CLUSTER_DEFAULT_CMD_OPTS_DEFAULT =
+        "-server -Djava.net.preferIPv4Stack=true -Dhadoop.metrics.log.level=WARN " +
+        "--add-opens java.base/java.lang=ALL-UNNAMED";
+    } else {
+      TEZ_AM_LAUNCH_CLUSTER_DEFAULT_CMD_OPTS_DEFAULT =
+        "-server -Djava.net.preferIPv4Stack=true -Dhadoop.metrics.log.level=WARN";
+    }
+  }
```

Comment:
This won't work in case someone configures the value of `TEZ_AM_LAUNCH_CLUSTER_DEFAULT_CMD_OPTS`, in that case default won't kick in itslef.

Should do the change in `TezClientUtils` in `constructAMLaunchOpts`

Reply:
Fixed